apex/contrib/csrc/bottleneck/bottleneck.cpp

```diff
@@ -0,0 +1,1634 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
```

Comment:
This is MIT license.  APEX should be under BSD 3-clause license https://github.com/NVIDIA/apex/blob/master/LICENSE .  (And a header like this should be in all the files, not just this one.)